### PR TITLE
[trivial] cleaning up code

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

1. Cleaning up events API
2. Removing dead code
3. Features use `enable` over a bit vector

### CODE REVIEW

File: types/src/on_chain_config/aptos_features.rs [hunk 1]
```diff
@@ -60,10 +60,20 @@ pub struct Features {
 
 impl Default for Features {
     fn default() -> Self {
```
Comment (igor-aptos) [inline types/src/on_chain_config/aptos_features.rs:62]: this should probably be removed , in favor of default_features(), @runtian-zhou  (for the separate PR)

**Review by igor-aptos — APPROVED** [review]

File: types/src/on_chain_config/aptos_features.rs (hunk 1 above)
Comment (runtian-zhou) [reply types/src/on_chain_config/aptos_features.rs:62]: I don't mind just merging it in. As long as we make it a separate commit message?

**Review by runtian-zhou — APPROVED** [review]

File: types/src/on_chain_config/aptos_features.rs (hunk 1 above)
Comment (igor-aptos) [reply types/src/on_chain_config/aptos_features.rs:62]: I mean it is more convoluted than just removing this default - as behavior of move unit tests will change, etc. 

So it's better for that to be done separately, and consequences adjusted

Comment (georgemitenkov) [reply types/src/on_chain_config/aptos_features.rs:62]: Let's do it separately then? Why do we need `default_features` over default impl?

Comment (runtian-zhou) [reply types/src/on_chain_config/aptos_features.rs:62]: Makes sense. We just need a centralized location for all the default features that will be enabled in the genesis

**Pull request merged**